pkg/proxy/envoyproxy.go

```diff
@@ -53,7 +59,15 @@ func createEnvoyRedirect(r *Redirect, wg *completion.WaitGroup) (RedirectImpleme
 	if envoyProxy != nil {
 		redir := &envoyRedirect{redirect: r}
 
-		envoyProxy.AddListener(r.id, r.ProxyPort, r.rules, r.ingress, redir, wg)
+		ip := r.source.GetIPv4Address()
+		if ip == "" {
+			ip = r.source.GetIPv6Address()
+		}
+		if ip == "" {
+			return nil, fmt.Errorf("%s: Cannot create redirect, proxy source has no IP address.", r.id)
+		}
+		envoyProxy.AddListener(r.id, ip, r.ProxyPort, r.ingress, redir, wg)
+		atomic.AddInt32(&redirectCount, 1)
```

Comment:
Since this variable is accessed across files, it might be worth wrapping the increment/get with functions. On the other hand, it's very clear what's going on now. shrug.

Reply:
an alternative would be to make it a custom type, and making them methods. Still more complexity, however

Reply:
Agreed. I plan to make it a private field in XDSServer in a subsequent PR.